cmd/tempo-vulture/main.go

```diff
@@ -299,3 +360,44 @@ func hasMissingSpans(t *tempopb.Trace) bool {
 
 	return false
 }
+
+func constructTraceFromEpoch(epoch time.Time) *tempopb.Trace {
+	r := newRand(epoch)
+	traceIDHigh := r.Int63()
+	traceIDLow := r.Int63()
+
+	trace := &tempopb.Trace{}
+
+	for i := int64(0); i < generateRandomInt(1, 100, r); i++ {
+		batch := makeThriftBatch(traceIDHigh, traceIDLow, r, epoch)
+		internalTrace := jaegerTrans.ThriftBatchToInternalTraces(batch)
+		conv, err := internalTrace.ToOtlpProtoBytes()
+		if err != nil {
+			logger.Error(err.Error())
+		}
+
+		t := tempopb.Trace{}
+		err = t.Unmarshal(conv)
+		if err != nil {
+			logger.Error(err.Error())
+		}
+
+		// Due to the several transforms above, some manual mangling is required to
+		// get the parentSpanID to match.  In the case of an empty []byte in place
+		// for the ParentSpanId, we set to nil here to ensure that the final result
+		// matches the json.Unmarshal value when tempo is queried.
+		for ib, b := range t.Batches {
+			for il, l := range b.InstrumentationLibrarySpans {
+				for is, s := range l.Spans {
+					if len(s.GetParentSpanId()) == 0 {
+						t.Batches[ib].InstrumentationLibrarySpans[il].Spans[is].ParentSpanId = nil
```

Comment:
Not sure I understand the need for array indexing here.  Could this be replaced with just `s.ParentSpanId = nil` ?

Reply:
I think if you are in the loop, you want to avoid referencing variables inside the loop as well when you intent to modify the variable outisde the loop, so here we take the index of each of the loop and make the change there.

Reply:
Ah good point.  I usually just modify the original out of habit.  I'll make the change.